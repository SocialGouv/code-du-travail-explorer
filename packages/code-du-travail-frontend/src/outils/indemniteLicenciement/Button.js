import React from "react";
import PropTypes from "prop-types";
const Button = props => {
  const { onPress, onClick, onKeyUp, children } = props;
  return (
    <div
      tabIndex="0"
      role="button"
      className="btn"
      onKeyDown={event => {
        /* Space & Enter */
        if (event.keyCode === 32 || event.keyCode === 13) {
          event.preventDefault();
          if (onPress) {
            onPress();
          }
          if (onClick) {
            onClick();
          }
        }
        if (onKeyUp) {
          onKeyUp(event);
        }
      }}
      onClick={event => {
        if (onPress) {
          onPress();
        }
        if (onClick) {
          onClick(event);
        }
      }}
    >
      {children}
    </div>
  );
};

Button.propTypes = {
  onPress: PropTypes.func,
  onKeyUp: PropTypes.func,
  onClick: PropTypes.func,
  style: PropTypes.object
};

export { Button };
