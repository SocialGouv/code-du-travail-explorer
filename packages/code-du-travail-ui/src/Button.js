import React from "react";
import PropTypes from "prop-types";

export const flavors = [
  "primary",
  "secondary",
  "warning",
  "success",
  "info",
  "danger"
];

// return the 1st truthy prop that is one of an Button versions
const getFlavor = props => {
  const versionProps = Object.keys(props).filter(
    // ensure value is truthy
    flavor => flavors.indexOf(flavor) > -1 && !!props[flavor]
  );
  return (versionProps.length && versionProps[0]) || "";
};

const Button = props => (
  <div className={`btn btn__${getFlavor(props)}`} {...props} />
);

Button.propTypes = {
  /** use .btn.btn__primary */
  primary: PropTypes.bool,
  /** use .btn.btn__secondary */
  secondary: PropTypes.bool,
  /** use .btn.btn__warning */
  warning: PropTypes.bool,
  /** use .btn.btn__success */
  success: PropTypes.bool,
  /** use .btn.btn__info */
  info: PropTypes.bool,
  /** use .btn.btn__danger */
  danger: PropTypes.bool,
  style: PropTypes.object
};

Button.defaultProps = {
  primary: false,
  secondary: false,
  warning: false,
  success: false,
  info: false,
  danger: false
};

export default Button;
