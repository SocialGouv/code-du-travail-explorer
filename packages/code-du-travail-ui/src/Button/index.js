import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { darken, lighten, transparentize } from "polished";

import { animations, box, colors, fonts, spacing, variants } from "../theme";

// We don't want the variant prop to be passed down to the button
// eslint-disable-next-line
const Button = ({ children, pressed, variant, onClick, ...props }) => {
  return (
    <button aria-pressed={pressed} {...props} onClick={onClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  variant: PropTypes.oneOf(["default", "icon", "link"].concat(variants)),
  onClick: PropTypes.func,
  pressed: PropTypes.bool
};

Button.defaultProps = {
  variant: "default",
  onClick: () => {},
  pressed: false
};

const StyledButton = styled(Button)`
  padding: ${spacing.small} ${spacing.base};
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: ${fonts.sizeBase};
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: ${box.borderRadius};
  cursor: pointer;
  transition: background-color ${animations.transitionTiming} ease;

  ${props => {
    let color = colors.primaryText;
    let backgroundColor = colors.blueLight;
    if (props.variant === "link") {
      return css`
        padding: 0;
        vertical-align: initial;
        line-height: initial;
        color: ${colors.blue};
        font-weight: normal;
        font-size: inherit;
        background: none;
        border: none;
        border-radius: 0;
        text-decoration: underline;
        &:focus,
        &:hover,
        &:active {
          text-decoration: none;
        }
      `;
    }
    if (props.variant === "icon") {
      return css`
        padding: ${spacing.base};
        color: ${colors.darkText};
        line-height: 0;
        border: none;
        &:hover {
          color: ${lighten(0.3, colors.darkText)};
        }
        &:active {
          position: relative;
          top: 1px;
        }
      `;
    }

    if (props.variant !== "default") {
      backgroundColor = colors[`${props.variant}Background`];
      color = colors[`${props.variant}Text`];
    }

    return css`
      color: ${color};
      background: ${backgroundColor};
      border-color: ${backgroundColor};
      border-bottom-color: ${darken(0.1, backgroundColor)};
      :not([disabled]) {
        &:hover,
        &:focus {
          background: ${lighten(0.05, backgroundColor)};
          color: ${lighten(0.05, color)};
        }
        &:active {
          color: ${lighten(0.1, color)};
          background: ${lighten(0.1, backgroundColor)};
          border-width: 2px 1px 1px 1px;
          border-color: ${backgroundColor};
          outline: none;
        }
      }
      &[aria-pressed="true"] {
        color: ${lighten(0.05, color)};
        background: ${lighten(0.05, backgroundColor)};
        border-width: 2px 1px 1px 1px;
        border-color: ${backgroundColor};
        border-top-color: ${darken(0.1, backgroundColor)};
        box-shadow: inset 0 1px 2px 0 ${darken(0.1, backgroundColor)};
        :not([disabled]) {
          &:hover,
          &:focus {
            background-color: ${backgroundColor};
          }
          &:active {
            border-top-color: ${darken(0.1, backgroundColor)};
          }
        }
      }
      /* keep it last so it overrides other styles */
      &[disabled] {
        cursor: not-allowed;
        color: ${transparentize(0.6, color)};
      }
    `;
  }}
`;

StyledButton.defaultProps = {
  variant: "default"
};

export default StyledButton;
