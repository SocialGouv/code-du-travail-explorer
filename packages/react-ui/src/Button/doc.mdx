---
name: Button
menu: Component
---

import { Playground, Props } from "docz";
import { X } from "react-feather";
import { Maximize } from "react-feather";
import Section from "../layout/Section"
import { variants } from "../theme";
import { Button } from ".";

# Button

## Standard buttons

<Playground>
  {
    () => ["default"].concat(variants).map((variant) => (
      <Section>
        <Button variant={variant}>{variant} button</Button>
        {" "}
        <Button variant={variant} disabled>{variant} button disabled</Button>
        {" "}
        <Button variant={variant} pressed>{variant} button pressed</Button>
      </Section>
    ))
  }
</Playground>

## Special buttons

<Playground>
  <Section>
    <Button variant="link">Button link</Button>
  </Section>
  <Section>
    <Button variant="icon">
      <X />
    </Button>
  </Section>
  <Section>
    <Button variant="icon">
      <Maximize />
    </Button>
  </Section>
</Playground>

<Props of={Button} />
