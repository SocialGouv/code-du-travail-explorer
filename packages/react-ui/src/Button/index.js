import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { ArrowRight } from "react-feather";
import { lighten, transparentize } from "polished";

import { animations, box, fonts, spacings } from "../theme";

export const StyledButton = styled.button`
  display: inline-flex;
  align-items: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: ${fonts.sizes.default};
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: ${box.borderRadius};
  cursor: pointer;
  transition: background-color ${animations.transitionTiming} linear,
    border-color ${animations.transitionTiming} linear, transform 100ms linear;
  appearance: none;

  ${({ narrow, theme, small, variant }) => {
    if (variant === "link") {
      return css`
        padding: 0;
        color: ${theme.primary};
        font-weight: 400;
        font-size: ${fonts.sizes.default};
        line-height: ${fonts.lineHeight};
        text-align: left;
        background: none;
        border: none;
        border-radius: 0;
        &:focus,
        &:hover,
        &:active {
          text-decoration: underline;
        }
      `;
    }

    let height = "5.2rem";
    let backgroundColor = theme[`${variant}`];
    let borderColor = theme[`${variant}`];
    let color = theme[`${variant}Text`];
    let boxShadow =
      "0px 10px 30px rgba(52, 77, 122, 0.26), 0px 4px 5px rgba(117, 152, 214, 0.35)";
    let opacity = "1";

    let padding = "0 4.4rem";

    if (small) {
      height = "4rem";
      padding = "0 3rem";
    }

    if (narrow) {
      padding = small ? "0 1rem" : "0 1.9rem";
    }

    if (variant === "flat") {
      color = theme.paragraph;
      backgroundColor = theme.white;
      borderColor = theme.border;
      boxShadow = "none";
    }

    if (variant === "naked") {
      color = theme.paragraph;
      backgroundColor = "transparent";
      borderColor = "transparent";
      boxShadow = "none";
      opacity = "0.6";
    }

    return css`
      height: ${height};
      padding: ${padding};
      color: ${color};
      background: ${backgroundColor};
      border-color: ${borderColor};
      box-shadow: ${boxShadow};
      opacity: 1;

      &:link,
      &:visited {
        text-decoration: none;
        color: ${color};
      }
      :not([disabled]) {
        &:hover,
        &:active,
        &:focus {
          opacity: ${opacity};
          transform: translateY(-2px);
          background: ${lighten(0.1, backgroundColor)};
          border-color: ${lighten(0.1, borderColor)};
        }
      }
      /* keep it last so it overrides other styles */
      &[disabled] {
        color: ${transparentize(0.6, color)};
        box-shadow: none;
        cursor: not-allowed;
      }
    `;
  }}
`;

const StyledArrowRight = styled(ArrowRight)`
  height: 1.4rem;
  margin-left: ${spacings.tiny};
  transition: transform ${animations.transitionTiming} linear;
  /* stylelint-disable-next-line */
  ${StyledButton}:hover & {
    transform: translateX(5px);
  }
`;

export const Button = React.forwardRef(({ children, ...props }, ref) => (
  <StyledButton {...props} ref={ref}>
    {children}
    {props.variant === "link" && <StyledArrowRight />}
  </StyledButton>
));
Button.displayName = "Button";

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  small: PropTypes.bool,
  narrow: PropTypes.bool,
  variant: PropTypes.oneOf(["link", "flat", "naked", "primary", "secondary"])
};

Button.defaultProps = {
  onClick: () => {},
  narrow: false,
  small: false,
  variant: "secondary"
};
