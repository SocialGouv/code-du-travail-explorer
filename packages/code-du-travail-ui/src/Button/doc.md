---
name: Button
menu: Component
---
import { Playground, Props } from "docz";
import IconClose from "react-feather/dist/icons/x";
import IconEnlarge from "react-feather/dist/icons/maximize";
import Button from ".";

# Button

<Playground>
  <p>
    <Button>Default button</Button>
    {" "}
    <Button variant="default" disabled>Default button disabled</Button>
    {" "}
    <Button pressed>Default button pressed</Button>
  </p>
  <p>
    <Button variant="primary">Button Primary</Button>
    {" "}
    <Button variant="primary" disabled>Button Primary disabled</Button>
    {" "}
    <Button variant="primary" pressed>Button Primary pressed</Button>
  </p>
  <p>
    <Button variant="secondary">Button Secondary</Button>
    {" "}
    <Button variant="secondary" disabled>Button Secondary disabled</Button>
    {" "}
    <Button variant="secondary" pressed>Button Secondary pressed</Button>
  </p>
  <p>
    <Button variant="info">Button Info</Button>
    {" "}
    <Button variant="info" disabled>Button Info disabled</Button>
    {" "}
    <Button variant="info" pressed>Button Info pressed</Button>
  </p>
  <p>
    <Button variant="success">Button Success</Button>
    {" "}
    <Button variant="success" disabled>Button Success disabled</Button>
    {" "}
    <Button variant="success" pressed>Button Succes pressed</Button>
  </p>
  <p>
    <Button variant="warning">Button Warning</Button>
    {" "}
    <Button variant="warning" disabled>Button Warning disabled</Button>
    {" "}
    <Button variant="warning" pressed>Button Warning pressed</Button>
  </p>
  <p>
    <Button variant="danger">Button Danger</Button>
    {" "}
    <Button variant="danger" disabled>Button Danger disabled</Button>
    {" "}
    <Button variant="danger" pressed>Button Danger pressed</Button>
  </p>
  <p>
    <Button variant="link">Button link</Button>
  </p>
  <p>
    <Button variant="icon">
      <IconClose />
    </Button>
    <br />
    <br />
    <Button variant="icon">
      <IconEnlarge />
    </Button>
  </p>
</Playground>

<Props of={Button} />
