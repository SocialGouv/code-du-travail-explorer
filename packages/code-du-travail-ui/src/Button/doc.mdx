---
name: Button
menu: Component
---

import { Playground, Props } from "docz";
import IconClose from "react-feather/dist/icons/x";
import IconEnlarge from "react-feather/dist/icons/maximize";
import Section from "../layout/Section"
import { variants } from "../theme";
import Button from ".";

# Button

## Standard buttons

<Playground>
  {
    () => ["default"].concat(variants).map((variant) => (
      <Section>
        <Button variant={variant}>{variant} button</Button>
        {" "}
        <Button variant={variant} disabled>{variant} button disabled</Button>
        {" "}
        <Button variant={variant} pressed>{variant} button pressed</Button>
      </Section>
    ))
  }
</Playground>

## Special buttons

<Playground>
  <Section>
    <Button variant="link">Button link</Button>
  </Section>
  <Section>
    <Button variant="icon">
      <IconClose />
    </Button>
  </Section>
  <Section>
    <Button variant="icon">
      <IconEnlarge />
    </Button>
  </Section>
</Playground>

<Props of={Button} />
