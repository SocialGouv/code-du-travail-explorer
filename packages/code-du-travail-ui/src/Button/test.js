import React from "react";
import { render } from "react-testing-library";
import Button from ".";

describe("<Button />", () => {
  it("renders", () => {
    const { container } = render(<Button>A button</Button>);
    expect(container).toMatchSnapshot();
  });

  test.each([
    ["default"],
    ["primary"],
    ["secondary"],
    ["info"],
    ["success"],
    ["warning"],
    ["danger"],
    ["icon"],
    ["link"]
  ])("it renders a Button %s", label => {
    const { container } = render(
      <Button variant={label}>this is a Button {label} </Button>
    );
    expect(container).toMatchSnapshot();
  });

  it("can be toggled programmatically", () => {
    const { getByText, rerender } = render(<Button pressed>label</Button>);
    const button = getByText("label");
    expect(button.getAttribute("aria-pressed")).toBe("true");
    rerender(<Button>label</Button>);
    expect(button.getAttribute("aria-pressed")).toBe("false");
  });
});
