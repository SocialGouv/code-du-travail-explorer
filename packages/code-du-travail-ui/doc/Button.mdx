---

---

import { Playground, PropsTable } from "docz";
import { Button } from "../src";

## \<Button/\>

<Playground>
  <Button>Button</Button>
  <br/><br/>
  <Button primary>Button primary</Button>
  <br/><br/>
  <Button secondary>Button secondary</Button>
  <br/><br/>
  <Button warning>Button warning</Button>
  <br/><br/>
  <Button success>Button success</Button>
  <br/><br/>
  <Button info>Button info</Button>
  <br/><br/>
  <Button danger>Button danger</Button>
</Playground>

<PropsTable of={Button} />