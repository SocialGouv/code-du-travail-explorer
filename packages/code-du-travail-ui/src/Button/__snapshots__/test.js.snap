// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it renders a Button danger 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #0c0c0e;
  background: #f6bcc2;
  border-color: #f6bcc2;
  border-bottom-color: #f08f99;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #f9d2d6;
  color: #18181c;
}

.c0:not([disabled]):active {
  color: #242429;
  background: #fce9eb;
  border-width: 2px 1px 1px 1px;
  border-color: #f6bcc2;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #18181c;
  background: #f9d2d6;
  border-width: 2px 1px 1px 1px;
  border-color: #f6bcc2;
  border-top-color: #f08f99;
  box-shadow: inset 0 1px 2px 0 #f08f99;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #f6bcc2;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #f08f99;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(12,12,14,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    danger
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button icon 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  padding: 1rem;
  color: #434956;
  line-height: 0;
  border: none;
}

.c0:hover {
  color: #8c94a6;
}

.c0:active {
  position: relative;
  top: 1px;
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    icon
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button info 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #0c0c0e;
  background: #d7e8f9;
  border-color: #d7e8f9;
  border-bottom-color: #abcff2;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #edf5fc;
  color: #18181c;
}

.c0:not([disabled]):active {
  color: #242429;
  background: #fff;
  border-width: 2px 1px 1px 1px;
  border-color: #d7e8f9;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #18181c;
  background: #edf5fc;
  border-width: 2px 1px 1px 1px;
  border-color: #d7e8f9;
  border-top-color: #abcff2;
  box-shadow: inset 0 1px 2px 0 #abcff2;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #d7e8f9;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #abcff2;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(12,12,14,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    info
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button link 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  padding: 0;
  vertical-align: initial;
  line-height: initial;
  color: #0053b3;
  font-weight: normal;
  font-size: inherit;
  background: none;
  border: none;
  border-radius: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus,
.c0:hover,
.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    link
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button primary 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #005994;
  border-color: #005994;
  border-bottom-color: #003a61;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #0068ae;
  color: #fff;
}

.c0:not([disabled]):active {
  color: #fff;
  background: #0078c7;
  border-width: 2px 1px 1px 1px;
  border-color: #005994;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #fff;
  background: #0068ae;
  border-width: 2px 1px 1px 1px;
  border-color: #005994;
  border-top-color: #003a61;
  box-shadow: inset 0 1px 2px 0 #003a61;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #005994;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #003a61;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    primary
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button secondary 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #0c0c0e;
  background: #eaeaea;
  border-color: #eaeaea;
  border-bottom-color: #d1d1d1;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #f7f7f7;
  color: #18181c;
}

.c0:not([disabled]):active {
  color: #242429;
  background: #fff;
  border-width: 2px 1px 1px 1px;
  border-color: #eaeaea;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #18181c;
  background: #f7f7f7;
  border-width: 2px 1px 1px 1px;
  border-color: #eaeaea;
  border-top-color: #d1d1d1;
  box-shadow: inset 0 1px 2px 0 #d1d1d1;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #eaeaea;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #d1d1d1;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(12,12,14,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    secondary
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button success 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #0c0c0e;
  background: #e0f2bd;
  border-color: #e0f2bd;
  border-bottom-color: #ccea92;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #eaf6d2;
  color: #18181c;
}

.c0:not([disabled]):active {
  color: #242429;
  background: #f4fae8;
  border-width: 2px 1px 1px 1px;
  border-color: #e0f2bd;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #18181c;
  background: #eaf6d2;
  border-width: 2px 1px 1px 1px;
  border-color: #e0f2bd;
  border-top-color: #ccea92;
  box-shadow: inset 0 1px 2px 0 #ccea92;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #e0f2bd;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #ccea92;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(12,12,14,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    success
     
  </button>
</div>
`;

exports[`<Button /> it renders a Button warning 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #0c0c0e;
  background: #fee5ad;
  border-color: #fee5ad;
  border-bottom-color: #fdd57b;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #feedc6;
  color: #18181c;
}

.c0:not([disabled]):active {
  color: #242429;
  background: #fff5df;
  border-width: 2px 1px 1px 1px;
  border-color: #fee5ad;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #18181c;
  background: #feedc6;
  border-width: 2px 1px 1px 1px;
  border-color: #fee5ad;
  border-top-color: #fdd57b;
  box-shadow: inset 0 1px 2px 0 #fdd57b;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #fee5ad;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #fdd57b;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(12,12,14,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    this is a Button 
    warning
     
  </button>
</div>
`;

exports[`<Button /> renders 1`] = `
.c0 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #006be6;
  border-color: #006be6;
  border-bottom-color: #0053b3;
}

.c0:not([disabled]):hover,
.c0:not([disabled]):focus {
  background: #07f;
  color: #fff;
}

.c0:not([disabled]):active {
  color: #fff;
  background: #1a85ff;
  border-width: 2px 1px 1px 1px;
  border-color: #006be6;
  outline: none;
}

.c0[aria-pressed="true"] {
  color: #fff;
  background: #07f;
  border-width: 2px 1px 1px 1px;
  border-color: #006be6;
  border-top-color: #0053b3;
  box-shadow: inset 0 1px 2px 0 #0053b3;
}

.c0[aria-pressed="true"]:not([disabled]):hover,
.c0[aria-pressed="true"]:not([disabled]):focus {
  background-color: #006be6;
}

.c0[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #0053b3;
}

.c0[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

<div>
  <button
    aria-pressed="false"
    class="c0"
  >
    A button
  </button>
</div>
`;
